pkg/kritis/crd/securitypolicy/violation.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package securitypolicy
+
+import (
+	"fmt"
+	"github.com/grafeas/kritis/pkg/kritis/apis/kritis/v1beta1"
+	"github.com/grafeas/kritis/pkg/kritis/constants"
+	"github.com/grafeas/kritis/pkg/kritis/metadata"
+)
+
+type violation int
+
+// A list of security policy violations
+// TODO: Add Attestation checking violations
+const (
+	UnqualifiedImageViolation violation = iota
+	FixesNotAvailableViolation
+	ExceedsMaxSeverityViolation
+)
+
+// SecurityPolicyViolation represents a vulnerability that violates an ISP
+type SecurityPolicyViolation struct {
+	Vulnerability metadata.Vulnerability
+	Violation     violation
+	Reason        string
+}
+
+// UnqualifiedImageViolationReason returns a detailed reason if the image is unqualified
+func UnqualifiedImageViolationReason(image string) string {
+	return fmt.Sprintf("%s is not a fully qualified image", image)
+}
+
+// FixesAvailableViolationReason returns a detailed reason if a CVE doesn't have a fix available
+func FixesNotAvailableViolationReason(image string, vulnz metadata.Vulnerability) string {
+	return fmt.Sprintf("found CVE %s in %s which doesn't have fixes available", vulnz.CVE, image)
```

Comment:
You need to reverse of this. I was also confused first. but the way this should be implemented is:

For Policy Spec, with OnlyFixesNotAvailable as false (default true) we allow images with vulnerabilities which have fixes out. If the fix is out, we want to encourage users to rebuild image and deploy with that image. Hence the default is set to true.
So the reason should be `found CVE %s in %s which has fixes available`
```
apiVersion: kritis.grafeas.io/v1beta1
kind: ImageSecurityPolicy
metadata:
  name: my-isp
  namespace: default
spec:
  imageWhitelist:
  - gcr.io/my/image
  packageVulnerabilityRequirements:
    maximumSeverity: HIGH
    onlyFixesNotAvailable: false
 
```

Reply:
Thanks for explaining this! :)